Readme adjustment (#147)
HackYourCareer/SmartKickers: README.md

> - Golang server for processing the data.
> - React client to display game information.

Reviewer: These are not sentences. Remove the periods at their ends.

```suggestion
- Golang server for processing the data
- React client to display game information
```

## Table of contents

> - [Overview](#overview)
> - [Table of content](#table-of-content)

Reviewer: No point keeping it here. It's already above the TOC, and right next to it, so you wouldn't use navigation to get there anyway, and TOC never self-refers. 
I am aware that you've already changed the TOC in the meantime, but this one actually makes more sense. 

```suggestion
```

- [Features](#features)
- [Prerequisites](#prerequisites)
- [Installation](#installation)
- [Launching](#launching)
  - [Launching the server and web application](#launching-the-server-and-web-application)
  - [Launching the ball-tracking software](#launching-the-ball-tracking-software)
- [Contact Information](#contact-information)

## Features

- Keeping track of the score
- Manually adjusting the score

## Prerequisites

Get the following:

1. [Git](https://git-scm.com/downloads)
2. [Python](https://www.python.org/downloads/) in version 3.10.5 or higher
3. [Node.js](https://nodejs.org/en/download/) in version 16.16.0 or higher
4. Access to [SmartKickersAI](https://github.com/HackYourCareer/SmartKickersAI)

   > **NOTE:** To get access to the SmartKickersAI repository, [contact](#contact-information) Team Beavers.

5. [Ximea camera drivers](https://www.ximea.com/support/wiki/apis/)
6. Docker daemon for example Docker Desktop:
   - [MacOS](https://docs.docker.com/desktop/install/mac-install/)
   - [Windows](https://docs.docker.com/desktop/install/windows-install/)
   - [Linux](https://docs.docker.com/desktop/install/linux-install/)
7. [NumPy](https://numpy.org/install/)

> > **TIP:** If you're a macOS user, you can also use `brew` to install NumPy:

Reviewer: This should be indented to fall under point 7. 
``````suggestion
  > **TIP:** If you're a macOS user, you can also use `brew` to install NumPy:
``````

> ```bash
> brew install numpy
> ```

Reviewer: This should be indented to fall under point 7. 
``````suggestion
  ```bash
  brew install numpy
  ```
``````

8. [OpenCV-Python](https://pypi.org/project/opencv-python/)

9. [Imutils](https://pypi.org/project/imutils/)

> Check if you have installed all of the [prerequisites](https://github.com/HackYourCareer/SmartKickers#prerequisites).

Reviewer: What's the point of this line? It refers you to the section you're just reading now. (I suspect it's a leftover from when you had two sections - Prerequisites and Dependencies, if I remember correctly). 
I'd remove it. 

```suggestion
```

> **TIP:** If you're a macOS user, you can run the [`requirementsCheck.sh`](requirementsCheck.sh) script to check for any missing dependencies. Note that the script looks for OpenCV, NumPy, Imutils, and the Ximea driver **only after you have successfully installed Python!** If you don't have Python, it doesn't print missing Python-dependent packages except Python itself.

> 
> ```bash
> sudo ./requirementsCheck.sh
> ```

Reviewer: This should be part of the TIP panel (under `>`).

```suggestion
> ```bash
> sudo ./requirementsCheck.sh
> ```
```

## Installation

1. Create a new directory and navigate to it in a terminal.

> ```bash
> mkdir smartkickers
> cd smartkickers
> ```

Reviewer: This should be indented to fall under 1. 

``````suggestion
  ```bash
  mkdir smartkickers
  cd smartkickers
  ```
``````

2. Clone the repository.

> ```bash
> git clone https://github.com/HackYourCareer/SmartKickersAI.git
> ```

Reviewer: This should be indented to fall under 2. 

``````suggestion
  ```bash
  git clone https://github.com/HackYourCareer/SmartKickersAI.git
  ```
``````

> 3. Pull frontend image from GCR

Reviewer: Definite article and period missing:
```suggestion
3. Pull the frontend image from GCR.
```

> ```bash
> docker pull ghcr.io/hackyourcareer/smartkickers-frontend:latest
> ```

Reviewer: This should be indented to fall under 3. 
``````suggestion
  ```bash
  docker pull ghcr.io/hackyourcareer/smartkickers-frontend:latest
  ```
``````

> 4. Pull backend image from GCR

Reviewer: Definite article and period missing:
```suggestion
4. Pull the backend image from GCR.
```

> ```bash
> docker pull ghcr.io/hackyourcareer/smartkickers-backend:latest
> ```

Reviewer: This should be indented to fall under 4. 
``````suggestion
  ```bash
  docker pull ghcr.io/hackyourcareer/smartkickers-backend:latest
  ```
``````

## Launching

> > **NOTE:** Configuration files for the camera view are provided by the developers in the [`SmartKickersAI`](https://github.com/HackYourCareer/SmartKickersAI/tree/main/LocalServer) repository.

Reviewer: I'd probably add "the" at the beginning, as they are specific files. 
```suggestion
> **NOTE:** The configuration files for the camera view are provided by the developers in the [`SmartKickersAI`](https://github.com/HackYourCareer/SmartKickersAI/tree/main/LocalServer) repository.
```

### Launching the server and web application

> - Run pulled containers
> 
>   - In a terminal run:

Reviewer: - An article is missing.
- If you have only one step, you don't need to format it as a list. Just type it as a regular sentence. 
  - However, the sub-elements are sequential steps (should be performed in a given order), so they should be an ordered (numbered) list. I don't think it's necessary to put them as sub-elements, as pulling the containers is all you have to do to launch. 
-  Because you have further subsections, consider adding info what you're launching here. (Like: `Run the pulled containers to launch the application interface.`)

```suggestion
Run the pulled containers.

1. In a terminal, run:
```

  ```bash
  docker run -p 3000:3000 ghcr.io/hackyourcareer/smartkickers-backend:latest
  ```

>   - Open a new terminal and run:

Reviewer: See the comment above.
```suggestion
2. Open a new terminal and run: 
```

  ```bash
  docker run -p 3007:80 ghcr.io/hackyourcareer/smartkickers-frontend:latest
  ```

### Launching the ball-tracking software

1. Plug in the camera to the computer.

> 2. In a new terminal window, launch the Node.js server.

Reviewer: I know you've changed this point in the meantime to: 
```
2. In order to launch the application, navigate to the `SmartKickersAI` repo and double click `start` file.
   Successful connection to the Go server shows the connection details like in the screenshot below:
   ![Node launch image](assets/nodeLaunch.png "Node launch")
```

So I'll refer to the updated one instead. 

- `In order to run` and `To run` express the same thing, so it's better to go with the simpler one (remember that your users are not native English speakers, so it's best that the language is simple and doesn't stand in the way of using your product.) 
- An article is missing before the file name. 
  - Also, **shouldn't there be an extension in this file name?** I can't check it 'cause I don't have access to that repo. 
- You shouldn't add screenshots with code snippets/terminal output to documentation. Your user might have the shell skin formatted differently to yours and the formatting might actually add cognitive load. Instead, copy the output and add it as a code block.
  - Also, is it still relevant? In the screenshot, you can see the command that you previously had in your documentation to launch the application, but now you click the `start` file. 

``````suggestion
2. To launch the application, navigate to the `SmartKickersAI` repo and double click the `start` file.
   On successful connection to the Go server, you see the following output:
  ```bash
    {
      url: "ws://127.0.0.1:3000",
      wsurl: "ws://127.0.0.1:3000/shot",
      dispatcherlURL: 'ws://127.0.0.1:3000'
      cfURL: 'ws://127.0.0.1:3000/cf'
      tableID: '10'
      token: 'c3d8c29ec43b9eadb8bc80ad1458ab8'
      messageTypes: {
          BallPosition: '615eec77d6be09356891'
          Shot: '2ca34aee4bb3dec060b1'
          BestShotGuid: 'c01b38461a678f3eefa5'
      } 
  }
  init
  conn!
  { type: 'utf8', utf8Data: '{"start":"10"}' }
  { start: '10' }
  game id
  10
  connected to IoT Services via wss
  ```
``````

Also, if you're the ones passing the URLs, I'd make the strings consistent and use either `'`s, or `"`s, not both.

> In order to launch the application, navigate to the `SmartKickersAI` repo and run:

Reviewer: See the comment above. 

```suggestion
```

> ```bash
> cd LocalServer/server
> node server.js
> ```

Reviewer: See the comment above. 

```suggestion
```

> Successful connection to the Go server shows the connection details like in the screenshot below:

Reviewer: See the comment above. 

```suggestion
```

> ![Node launch image](assets/nodeLaunch.png "Node launch")

Reviewer: See the comment above. 

```suggestion
```

> 3. In your Internet browser, go to [`localhost:3007`](http://localhost:3007/) to see the React application.

Reviewer: `to open the React application`?
Or `to navigate to the React application`. 

```suggestion
3. In your Internet browser, go to [`localhost:3007`](http://localhost:3007/) to open the React application.
```

> This takes you to the SmartKickers React application.

Reviewer: You already added it to the previous sentence, so this sentence doesn't bring in anything new to the table. 
But it's good that you're ending the instructions with the expected result. 
I propose that you change it to something like: 
```suggestion
You can now play the SmartKickers game. Enjoy!
```

- Golang server for processing the data.
- React client to display game information.

> ## Table of content

Author: Changed

- [SmartKickers](#smartkickers)
  - [Overview](#overview)
  - [Table of content](#table-of-content)
  - [Features](#features)
  - [Prerequisites](#prerequisites)
  - [Installation](#installation)
  - [Launching](#launching)
    - [Launching the server and web application](#launching-the-server-and-web-application)

>     - [Launching the ball-tracking software](#launching-the-ball-tracking-software)

Author: Changed. Now our TOC doesn't include a title.

## Features

- Keeping track of the score.

> - Manually adjust the score.

Author: Changed

## Prerequisites

> Install the following:

Author: Changed

1. [Git](https://git-scm.com/downloads)
2. [Python](https://www.python.org/downloads/) in version 3.10.5 or higher

> 3. [Node](https://nodejs.org/en/download/) in version 16.16.0 or higher

Author: Changed

> 4. Access to [SmartKickersAI](https://github.com/HackYourCareer/SmartKickersAI)

Author: Changed

>    **NOTE:** To get access for SmartKickersAI repository contact someone from [Team Beavers](https://wiki.one.int.sap/wiki/display/saplabspoland/Beavers+Team) on slack or via email.

Author: Changed. We have created a separate contact section with our e-mails and links to github profiles.

5. [Ximea camera drivers](https://www.ximea.com/support/wiki/apis/)

> 6. Docker API for example Docker Desktop:

Author: We will double check it.

   - [MacOS](https://docs.docker.com/desktop/install/mac-install/)
   - [Windows](https://docs.docker.com/desktop/install/windows-install/)
   - [Linux](https://docs.docker.com/desktop/install/linux-install/)

> 7. Numpy

Author: Changed

   ```bash
   pip3 install numpy

>    ```

Author: Changed

>    For **brew** users:

Author: Changed

   ```bash
   brew install numpy

>    ```

Author: Changed

> 8. OpenCV-python

Author: Changed

   ```bash
   pip3 install opencv-python

>    ```

Author: Changed

   For **brew** users:

   ```bash
   brew install opencv

>    ```

Author: It should end in `opencv` but we don't mention this command in the instruction anymore.

9. Imutils

```bash
pip3 install imutils

> ```

Author: Changed

## Installation

> 1. Make a new directory and open it in command line terminal.

Author: Changed

   ```bash
   mkdir smartkickers
   cd smartkickers
   ```

2. Clone the repositories.

   ```bash
   git clone https://github.com/HackYourCareer/SmartKickers.git
   git clone https://github.com/HackYourCareer/SmartKickersAI.git
   ```

3. Check if you have installed all of the [prerequisites](https://github.com/HackYourCareer/SmartKickers#prerequisites).

>    **MacOS** users can run the [`requirementsCheck.sh`](requirementsCheck.sh) script to check if they have any missing dependencies.

Author: Changed

   **NOTE:** The script will look for open-cv, numpy, imutils, ximer driver **only after successfully installing python!** So if you don't have python it won't print missing python dependent packages except python itself.

   ```bash

>    cd SmartKickers

Author: Changed

   sudo ./requirementsCheck.sh

>    ```

Author: Changed

>    **Windows** users have to check the dependencies manually.

Author: The script should also work on Linux but we have not tested it yet, so we mention only macOS.

> 4. Install missing dependencies. The links can be found in [prerequisites section.](https://github.com/HackYourCareer/SmartKickers#prerequisites)

Author: Changed

> 5. If you you use Docker, in folder [`SmartKickers`](SmartKickers) run docker-compose build.

Author: We have changed the instruction. Now we require docker but the user doesn't have to build an image.

> > **NOTE:** Configuration files for the camera view are provided by the developers in [`SmartKickersAI`](https://github.com/HackYourCareer/SmartKickersAI/tree/main/LocalServer) repository.

Author: Changed

### Launching the server and web application

- In the folder **SmartKickers** launch the docker-compose.

  ```bash
  docker-compose up
  ```

  After launching you will see that the backend and the frontend of the application have started successfully.
  ![App launch image](assets/appLaunch.png "App launch")

### Launching the ball-tracking software

> - Plug-in the camera wire to the computer.

Author: Changed

> - In new terminal switch to [`SmartKickersAI/LocalServer/server`](SmartKickersAI/LocalServer/server) and launch the node server.

Author: Changed

  ```bash

>   cd SmartKickersAI/LocalServer/server

Author: Changed

  node server.js
  ```

>   You will see the following output indicating that we have successfully connected to the go server.

Author: Changed, but we have stayed with the screenshot.

  ![Node launch image](assets/nodeLaunch.png "Node launch")

> - In the browser, go to [`localhost:3007`](http://localhost:3007/) to see the react application.

Author: On this stage, the app is already running in terminal and it's just a dashboard.

>   Properly working React application looks like this.

Reviewer: ```suggestion
  This takes you to the SmartKickers React application.
```

Author: Changed
